### ISSUE TITLE
[ARM32] Assertion failed 'type == TYP_DOUBLE || type == TYP_FLOAT' in 'HFATest.TestMan:Identity_HFA01(struct):struct'

### ISSUE THREAD

**Issue by alpencolt** [dotnet/coreclr#12855]
Crash on Ryu and legacy JIT:
```
./corerun ../tests-checked/JIT/jit64/hfa/main/testE/hfa_nf0E_d/hfa_nf0E_d.exe 

Assert failure(PID 1040 [0x00000410], Thread: 1040 [0x0410]): Assertion failed 'type == TYP_DOUBLE || type == TYP_FLOAT' in 'HFATest.TestMan:Identity_HFA01(struct):struct' (IL size 2)

    File: /home/alexander/src/coreclr/src/jit/instr.cpp Line: 3684
    Image: /home/asoldatov/src/corerun

Aborted
```

`../tests-checked/JIT/jit64/hfa/main/testE/hfa_nf0E_r/hfa_nf0E_r.exe` test version works well.
Bassed on commit:
```
commit 5ceac6362ffbc61b5159e4413d678a8ff6176f8b
Author: Ruben Ayrapetyan <ruben-ayrapetyan@users.noreply.github.com>
Date:   Mon Jul 17 17:10:42 2017 +0300

    Fix alignment of reads in MD5Transform. (#12800)
```

cc @dotnet/arm32-contrib

**Comment by BruceForstall** [comment]
This is a regression due to my change #12813. I'll have to find code that works for all cases.

**Comment by BruceForstall** [comment]
Fixed with #12871
